### PR TITLE
vm::utils::ascii! -> common::ascii!

### PR DESCRIPTION
**Pull request by youknowone**

### CODE REVIEW

File: common/src/str.rs [hunk 1]
```diff
@@ -82,6 +82,28 @@ pub fn to_ascii(value: &str) -> String {
     ascii
 }
 
+pub const fn bytes_is_ascii(x: &str) -> bool {
```
Comment (coolreader18) [inline common/src/str.rs:85]: This could be `#[doc(hidden)]`, since it's an implementation detail

**Review by coolreader18 — APPROVED** [review]

File: common/src/str.rs [hunk 2]
```diff
@@ -82,6 +82,29 @@ pub fn to_ascii(value: &str) -> String {
...
+    true
+}
+
+#[macro_export]
+macro_rules! ascii {
+    ($x:literal) => {{
+        const _: () = {
+            ["not ascii"][!::rustpython_common::str::bytes_is_ascii($x) as usize];
+        };
+        unsafe { ::ascii::AsciiStr::from_ascii_unchecked($x.as_bytes()) }
```
Comment (coolreader18) [inline common/src/str.rs:104]: Oh wait sorry this should probably also be a `$crate::` reference and the `ascii` crate should be `pub use`'d somewhere

File: common/src/str.rs [hunk 3]
```diff
@@ -82,6 +82,29 @@ pub fn to_ascii(value: &str) -> String {
...
+        i += 1;
+    }
+    true
+}
+
+#[macro_export]
+macro_rules! ascii {
+    ($x:literal) => {{
+        const _: () = {
+            ["not ascii"][!::rustpython_common::str::bytes_is_ascii($x) as usize];
```
Comment (coolreader18) [inline common/src/str.rs:102]: This can be `$crate::str::...`

File: common/src/str.rs (hunk 2 above)
Comment (youknowone) [reply common/src/str.rs]: I created `common::vendored::ascii`

**Pull request merged**